### ISSUE TITLE
azurerm_kubernetes_cluster_node_pool forces replacement due to ID case change

### ISSUE THREAD

**Issue by zerodayyy** [hashicorp/terraform-provider-azurerm#14746]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v1.1.2
azurerm v2.90.0

### Affected Resource(s)

* `azurerm_kubernetes_cluster_node_pool `

### Terraform Configuration Files

```hcl
resource "azurerm_kubernetes_cluster_node_pool" "workers" {
  for_each = var.worker_node_pools

  name                  = each.key
  kubernetes_cluster_id = data.azurerm_kubernetes_cluster.kubernetes.id
  vm_size               = each.value.vm_size
  priority              = each.value.priority
  mode                  = "User"
  orchestrator_version  = coalesce(each.value.kubernetes_version, data.azurerm_kubernetes_cluster.kubernetes.kubernetes_version)
  os_disk_size_gb       = each.value.os_disk_size_gb
  node_count            = each.value.node_count
  enable_auto_scaling   = each.value.autoscaling
  min_count             = each.value.min_count
  max_count             = each.value.max_count

  node_labels = lookup(each.value, "node_labels", {})

  lifecycle {
    ignore_changes = [eviction_policy, node_count, node_taints, vnet_subnet_id, node_labels, os_disk_size_gb]
  }
}
```

### Expected Behaviour

State of `azurerm_kubernetes_cluster_node_pool` resource can be moved without any problems
 
### Actual Behaviour

When moving state or re-importing `azurerm_kubernetes_cluster_node_pool` resource, it forces replacement on `kubernetes_cluster_id` property, the change being following:

```
"/subscriptions/xxxxxxxxxxxx-xxxx-xxxx-xxxxxxx/resourcegroups/xxxxxxxx/providers/Microsoft.ContainerService/managedClusters/cluster-name" -> "/subscriptions/xxxxxxxxxxxx-xxxx-xxxx-xxxxxxx/resourceGroups/xxxxxxxx/providers/Microsoft.ContainerService/managedClusters/cluster-name" # forces replacement
```

Note the difference in `resourcegroups` -> `resourceGroups`.

### Steps to Reproduce

1. `terraform apply`
2. `terraform state rm azurerm_kubernetes_cluster_node_pool.workers`
3. `terraform import azurerm_kubernetes_cluster_node_pool.workers '/subscriptions/xxxxxxxxxxxx-xxxx-xxxx-xxxxxxx/resourcegroups/xxxxxxxx/providers/Microsoft.ContainerService/managedClusters/cluster-name'`
4. `terraform plan` and observe the described behavior

### References

This issue was reported 6 months ago on StackOverflow [here](https://stackoverflow.com/questions/67825862/terraform-forces-aks-node-pool-replacement-without-any-changes).

**Comment by cberge908** [comment]
This is still present in azurerm 3.x (tested with 3.0.2).

**Comment by tombuildsstuff** [comment]
hi @zerodayyy 

Thanks for opening this issue.

Taking a look through here the issue is that the Resource ID being provided at Import time doesn't match what Terraform is expecting, [as per the documentation](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/kubernetes_cluster_node_pool#import) Terraform expects a Resource ID in the format (note the upper-case `G` in `resourceGroups`):

> /subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/group1/providers/Microsoft.ContainerService/managedClusters/cluster1/agentPools/pool1

Whilst we do validate the Resource ID matches what we're expecting during import, we aren't validating the casing of the `resourceGroups` segment in every resource at this time, but this is something we'll be doing in the not-too-distant future.

Since this can be fixed by specifying the Resource ID formatting that Terraform is expecting, I'm going to close this issue for the moment - however the issue with the casing of the `resourceGroups` not being validated will be fixed during the 3.0 provider lifecycle.

Thanks!
